### PR TITLE
Fix regressions on secure boot #125

### PR DESCRIPTION
**Pull request by gaojiaqi7**

### CODE REVIEW

File: td-shim/src/bin/td-shim/main.rs [hunk 1]
```diff
@@ -540,3 +511,32 @@ fn prepare_hob_list(
 
     hob_base
 }
+
+fn verify_payload<'a>(payload: &'a [u8], td_event_log: &mut TdEventLog) -> &'a [u8] {
+    #[cfg(feature = "secure-boot")]
+    {
+        let cfv = memslice::get_mem_slice(memslice::SliceType::Config);
+        let verifier = verifier::PayloadVerifier::new(payload, cfv);
```
Comment (jiangliu) [inline td-shim/src/bin/td-shim/main.rs:519]: using `expect!("Secure Boot: Cannot read verify header from payload binary");` may simplify the code here.

File: td-shim/src/bin/td-shim/main.rs [hunk 2]
```diff
@@ -149,6 +149,7 @@ pub extern "win64" fn _start(
         pi::fv::SECTION_PE32,
     )
     .expect("Failed to get image file from Firmware Volume");
+    payload = verify_payload(payload, &mut td_event_log);
```
Comment (jiangliu) [inline td-shim/src/bin/td-shim/main.rs:152]: Using `#[cfg(feature = "secure-boot")]` may help to simplify implementation of `verify_payload()`. It also help to rename it as `secure_boot_verify_payload()`?

Comment (gaojiaqi7) [reply td-shim/src/bin/td-shim/main.rs]: Thanks for your suggestions 👍

**Review by jyao1 — APPROVED** [review]

File: td-shim/src/bin/td-shim/main.rs [hunk 3]
```diff
@@ -540,3 +516,28 @@ fn prepare_hob_list(
...
+        EV_PLATFORM_CONFIG_FLAGS,
+        b"td payload",
+        verifier::PayloadVerifier::get_trust_anchor(cfv).unwrap(),
+    );
+    verifier.verify().expect("Verification fails");
+    td_event_log.create_event_log(4, EV_PLATFORM_CONFIG_FLAGS, b"td payload", payload);
+    td_event_log.create_event_log(
+        4,
+        EV_PLATFORM_CONFIG_FLAGS,
+        b"td payload svn",
```
Comment (jodh-intel) [inline td-shim/src/bin/td-shim/main.rs:537]: Nit: You could consider creating `const`s for `b"td payload"` and variants.

**Review by jodh-intel — APPROVED** [review]
Thanks @gaojiaqi7.

**Comment by jyao1** [comment]
@jiangliu , any more feedback?
I hope to merge it soon since it is a regression.

**Pull request merged**